Breaching charge

Diegoflores31 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
**Breaching Charge**
This Plastic explosive variant becomes a handy tool in quick assault operations featuring a faster planting time and smaller blast to quickly enter into fortified locations while keeping the operator safe.

After a fast planting this explosive will destroy the walls adjacent to it (half the damage as C4 and doesnt break reinforced walls) releasing deadly shrapnel at the opposite side of the wall causing damage to the enemies behind of it and providing a hole for the operators to start an assault.

**Limitations**
-cannot be planted on mobs
-cannot be planted on objects (including Burrower holes)
-no explosion damage/stun
-less damage than C4 to structures

**Pros**
-faster planting time
-faster explosion time
-shrapnel provides a good amount of damage to the target behind the wall
-safer since it uses a directional explosion

**Locations**
-Can be bought at Engie and SL vendors for 5 and 7 points
-Req Vendors will have some of them on stock each round
-Req can buy a crate of Breaching charges via ASRS

**concept video (final product may be a little different)**

https://user-images.githubusercontent.com/47069269/168969024-5c92e5b5-ba2a-42fa-a106-7c0bb00034fb.mp4




## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. --> Provides a new Tool for engineers in order to quickly breach into enemy structures with the added benefit of not killing the operators .
this is designed to compete with the C4 and Underbarrel shotgun to quickly break into the front.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Adds Breaching Charge . A fast planting explosive that keeps the operator safe while emiting deadly shrapnel towards the target . This new explosive is available at Squad Engineer , SL and REQ vendors.
add: Adds Breaching Charge crate to ASRS

/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

realforest2001 commented:
You'll need to update your changelog to the new format.

Agameofscones reviewed (changes requested):
you've been cursed by html/lengord_pouches.yml. You need to rebase onto the latest copy of master and discard / delete that file. Otherwise the sprites look good to me.

Diegoflores31 reviewed (commented):
file deleted . i guess im free from the curse now

Diegoflores31 reviewed (commented):
.

stanalbatross commented:
initial feedback so far seems to be that they take too long to detonate and are not useful enough. 

also fill out a more detailed changelog

Katskan commented:
I never saw anyone using them. Possibly having a faster plant time would give them an edge, most people I saw were just talking about them as "worse C4"

c4xmaniac commented:
I didn’t play much during the TM, but I never heard anyone using them and like Katskan said it seems to just be a worse C4. This being in combination with an intended master key nerf going in along side this MR, the plant time for the breaching charges should be decreased dramatically along with its point cost. I’m pretty sure I can buy c4 for the same price and it does basically the same thing with the addition of being able to blow tunnels.

somepersondunno commented:
Make breaching charges cost 3 and regular Plastic Explosives cost 5 and this will be a lot more viable. I had a lot of fun playing engineer and using these to destroy walls and, as the name implies, BREACH into enemy defenses. Additionally, making them take a little less time to set up would be nice, but overall a lot of fun IMO.

IngloriouslyBad commented:
Biggest problem with C4 and the new breaching charge is the set up times for both are just too long. Xenos aren't going to be surprised ever by C4 or breaching charges when you have to stand still ominously against a wall with a big red ring above your head, yelling C4 C4 C4. Xenos point at the wall before you're even done placing and get out of dodge or set up to counter you with the time it takes to set up and then go off. And good luck using either to widen a pre existing gap, a xeno will walk around the corner and push you 1 tile, stun you through the wall like you see with frontline burrowers, or just kill you in the case of T3s who turn the corner and tear you a new one, and retreat back inside.

Sargeantmuffinman commented:
Just from arm chair thoughts, reading other comments and me testing this thing a couple of times in the T-Dome.
Too long to plant, xenos knowing exactly what's going to happen due to 2 red circles over a marine and cost.

Frankly if you addressed the planting time and the cost, it'd make it more worth while.
My two cents.
Cent 1: Planting time is 2 seconds.
Cent 2: Cost is either reduced or the breaching charge comes in stacks of 2 or 3 so you can plant multiple.

Diegoflores31 commented:
added faster planting time
added a new sharpnel type with a more appealing view than the standard
fixed some issues with the explosion eating a good amount of shrapnell
reduced amount of shrapnel but increased the damage
Adjusted Shrapnel spawn turf to avoid collitions with the wall

Nanu308 commented:
Balance lgtm, might need supply cost tweaks later down the line but we'll see.

Geevies commented on code/game/objects/items/explosives/plastic.dm:
```diff
@@ -14,10 +14,13 @@
 	var/atom/plant_target = null //which atom the plstique explosive is planted on
 	var/overlay_image = "plastic-explosive2"
 	var/image/overlay
+	var/breachable = list(/obj/structure/window, /turf/closed, /obj/structure/machinery/door, /obj/structure/mineral_door)
```
```suggestion
	var/list/breachable = list(/obj/structure/window, /turf/closed, /obj/structure/machinery/door, /obj/structure/mineral_door)
```

Geevies commented on code/datums/supply_packs/weapons.dm:
```diff
@@ -246,6 +246,17 @@
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge,
+					/obj/item/explosive/plastic/breaching_charge)
+	cost = RO_PRICE_NORMAL
```
```suggestion
					/obj/item/explosive/plastic/breaching_charge
	)
	cost = RO_PRICE_NORMAL
```
